pkg/provisioner/ironic/ironic.go

```diff
@@ -354,6 +354,39 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 
 }
 
+func (p *ironicProvisioner) updateAutomatedClean(ironicNode *nodes.Node) (err error) {
+	p.log.Info("setting cleaning mode",
+		"ID", ironicNode.UUID,
+		"mode", p.host.Spec.AutomatedCleaningMode)
+
+	p.publisher("CleaningMode", string(p.host.Spec.AutomatedCleaningMode))
```

Comment:
Not sure if there's a problem in my env but this resulted being quite verbose for me:

![DeepinScreenshot_select-area_20210324082825](https://user-images.githubusercontent.com/13816/112278594-eae97800-8c7a-11eb-959b-0381c8ae709e.png)

Reply:
Seems that's due the publisher is being called on all the states if Ironic node. 
@hardys do you think it is okay that I add the publisher (for deprovisioning) when Ironic node is either in [`Active or DeployFail`](https://github.com/metal3-io/baremetal-operator/blob/5ebf0ab6e367a854c4177edc6612381b1d65dc01/pkg/provisioner/ironic/ironic.go#L1600)  or maybe when it is [`deleting`](https://github.com/metal3-io/baremetal-operator/blob/5ebf0ab6e367a854c4177edc6612381b1d65dc01/pkg/provisioner/ironic/ironic.go#L1584) ?

And during provisioning, I was thinking maybe to add publisher when Ironic node is in [Active](https://github.com/metal3-io/baremetal-operator/blob/5ebf0ab6e367a854c4177edc6612381b1d65dc01/pkg/provisioner/ironic/ironic.go#L1477)?

Reply:
@fmuyassarov do we need an event for this at all?

I'm thinking we don't output events for many other types of configuration, so we could just move this update to `ValidateManagementAccess()` as @zaneb mentioned, and apend it to the [other updates](https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L472..L479)?  Maybe log something but drop the event output?

Reply:
I added events here as per https://github.com/metal3-io/baremetal-operator/pull/784#discussion_r585064586. I can replace events with simple logging if that's okay.

Reply:
I have removed events, and instead kept only logging that was on top of that. I hope that's ok ?
```
	p.log.Info("setting automated cleaning mode to",
		"ID", ironicNode.UUID,
		"mode", p.host.Spec.AutomatedCleaningMode)
```

Reply:
@hardys  Isn't `ValidateManagementAccess` is used only when we are registering the host for the first time? If yes, that will work when someone enables/disables cleaning before provisioning. But what if we want to enable/disable cleaning before deprovisioning?

Reply:
@zaneb any thoughts on above question?

Reply:
`ValidateManagementAccess()` is called in ~every state since #716.

Reply:
> I added events here as per [#784 (comment)](https://github.com/metal3-io/baremetal-operator/pull/784#discussion_r585064586). I can replace events with simple logging if that's okay.

Logging wfm - my previous comment was just to say that I did generate an event in the inspection-disabled case - but if we generate an event (or log output) we want to do that once, not repeatedly